src/domace/ju1ce/PasovneMatrike.jl

```diff
@@ -0,0 +1,454 @@
+    end
+    if i > 0 && i <= size(A.s)[1]
+        return A.s[i][I[1]-i]
+    end
+    return 0
+
+end
+
+"""
+    setindex(A,v,I) ali A[i,j] = v
+
+    Nastavi element na mestu i, j na v, ce je to mesto v pasu, drugače ne naredi ničesar.
+
+"""
+
+function setindex!(A::SpodnjePasovnaMatrika, v, I::Vararg{Int,2})
+    
+    if I > size(A)
+        return
+    end
+    
+    i =  I[1] - I[2]
+
+    if i == 0
+        A.d[I[1]] = v
+    end
+    if i > 0 && i <= size(A.s)[1]
+        A.s[i][I[1]-i] = v
+    end
+end
+
+import Base:*, \
+import LinearAlgebra:lu
+
+"""
+    A*v
+    Množenje pasovne matrike A s vektorjem v
+"""
+
+function *(A::PasovnaMatrika, v::Vector)
+    y = zeros(size(A.d))
+    
+    for i = 1 : size(A.d)[1]
+       y[i] = v[i] * A.d[i]
+    end
+    
+    for i = 1 : size(A.s)[1]
+       for j = 1 : size(A.s[i])[1]
+            y[j+i] += v[j] * A.s[i][j]
+        end
+    end
+    
+    for i = 1 : size(A.z)[1]
+       for j = 1 : size(A.z[i])[1]
+            y[j] += v[j+i] * A.z[i][j]
+        end
+    end
+    
+    return y
+end
+
+"""
+    A*v
+    Množenje zgornje pasovne matrike A s vektorjem v
+"""
+
+function *(A::ZgornjePasovnaMatrika, v::Vector)
+    y = zeros(size(A.d))
+    
+    for i = 1 : size(A.d)[1]
+       y[i] = v[i] * A.d[i]
+    end
+    
+    for i = 1 : size(A.z)[1]
+       for j = 1 : size(A.z[i])[1]
+            y[j] += v[j+i] * A.z[i][j]
+        end
+    end
+    
+    return y
+end
+
+"""
+    A*v
+    Množenje spodnje pasovne matrike A s vektorjem v
+"""
+
+function *(A::SpodnjePasovnaMatrika, v::Vector)
+    y = zeros(size(A.d))
+    
+    for i = 1 : size(A.d)[1]
+       y[i] = v[i] * A.d[i]
+    end
+    
+    for i = 1 : size(A.s)[1]
+       for j = 1 : size(A.s[i])[1]
+            y[j+i] += v[j] * A.s[i][j]
+        end
+    end
+    
+    return y
+end
+
+"""
+    L, U = lu(A)
+    Izračuna LU razcep, če je matrika diagonalno dominantna.
+    L je tipa SpodnjeTrikotnaMatrika, U pa ZgornjeTrikotnaMatrika
+"""
+
+function lu(A::PasovnaMatrika)
+    
+    #test whether matrix is diagonaly dominant
+    for i = 1:length(A.d)   
+        @assert 2*A[i,i] >= sum(A[i,:])     #matrix is not diagonaly dominant!
+    end
+    
+    n = length(A.d)
+    ns = length(A.s)
+    nz = length(A.z)
+    
+    U = PasovnaMatrika(deepcopy(A.d),deepcopy(A.s),deepcopy(A.z))
+    #U = copy(A)
+    L = SpodnjePasovnaMatrika(ones(length(A.d)),deepcopy(A.s))
+
+    #Calculate L and U through gauss elimination
+    for i = 1:n-1
+        for j = i+1:i+ns
+            if j > n
+                break
+            end
+            l = U[j,i]/U[i,i]
+            L[j,i] = l
+            U[j,:] -= U[i,:]*l
+        end
+    end
+
+    U = ZgornjePasovnaMatrika(U.d,U.z)
+    
+    return L, U
+end
+
+"""
+    x = A\b
+    Izračuna sistem A*x=b
+"""
+
+function \(A::PasovnaMatrika, b::Vector)
+     
+    x = copy(b)
+    n = length(b)
+    ns = length(A.s)
+    nz = length(A.z)
+
+    #gauss elimination
```

Comment:
Uporabite raje `lu` razcep in operatorje `\` za `ZgornjePasovnaMatrika` in `SpodnjePasovnaMatrika`, ki ste ga že tako ali tako implementirali.